lib/util/http/github.ts

```diff
@@ -180,6 +182,81 @@ function constructAcceptString(input?: any): string {
   return acceptStrings.join(', ');
 }
 
+const MAX_GRAPHQL_PAGE_SIZE = 100;
+
+function getGraphqlPageSize(
+  fieldName: string,
+  defaultPageSize = MAX_GRAPHQL_PAGE_SIZE
+): number {
+  const cache = getCache();
+  cache.platform ??= {};
+  cache.platform.github ??= {};
+  cache.platform.github.graphqlPageCache ??= {};
+  const cachedRecord = cache.platform.github.graphqlPageCache[fieldName];
+
+  if (cachedRecord) {
+    logger.debug(
+      { fieldName, ...cachedRecord },
+      'GraphQL page size: found cached value'
+    );
+
+    const oldPageSize = cachedRecord.pageSize;
+
+    const now = DateTime.local();
+    const then = DateTime.fromISO(cachedRecord.pageLastResizedAt);
+    const expiry = then.plus({ hours: 24 });
+    if (now > expiry) {
+      const newPageSize = Math.min(oldPageSize * 2, MAX_GRAPHQL_PAGE_SIZE);
+      if (newPageSize < MAX_GRAPHQL_PAGE_SIZE) {
+        const timestamp = now.toISO();
+
+        logger.debug(
+          { fieldName, oldPageSize, newPageSize, timestamp },
+          'GraphQL page size: expanding'
+        );
+
+        cache.platform.github.graphqlPageCache[fieldName] = {
+          pageLastResizedAt: timestamp,
+          pageSize: newPageSize,
+        };
```

Comment:
I wonder if we need to set this during the `get`? Won't it be written later on by the `set` function anyway?

Reply:
We expand page size twice before any processing depending on 24-hours timeout, and shrink it after processing depending on errors received. It's possible to extract page expanding logic from `getGraphqlPageSize()` function, but they would go together anyway, and will share similar code branch structure 🤷‍♂️